portfolio/src/main/java/com/google/sps/data/FullBook.java

```diff
@@ -0,0 +1,171 @@
+  public String thumbnail;
+  public String isbn; // stores isbn 13 value
+
+  public FullBook(String title, ArrayList<String> authors, String isbn, String description) {
+    this.title = title;
+    this.isbn = isbn;
+    this.authors = authors;
+    this.description = description;
+  }
+
+  // for tester
+  public FullBook(String title, String language, String pageCount, String publishedDate,
+      String publisher, String maturityRating, String infoLink, String thumbnail,
+      ArrayList<String> authors, String isbn) {
+    this.title = title;
+    this.isbn = isbn;
+    this.authors = authors;
+    this.language = language;
+    this.pageCount = pageCount;
+    this.publishedDate = publishedDate;
+    this.publisher = publisher;
+    this.maturityRating = maturityRating;
+    this.infoLink = infoLink;
+    this.thumbnail = thumbnail;
+  }
+
+  public String getStringField(String fieldName) throws NoSuchFieldException {
+    switch (fieldName) {
+      case "title": {
+        return this.title;
+      }
+      case "genre": {
+        return this.title;
+      }
+      case "language": {
+        return this.language;
+      }
+      case "description": {
+        return this.description;
+      }
+      case "infoLink": {
+        return this.infoLink;
+      }
+      case "pageCount": {
+        return this.pageCount;
+      }
+      case "publishedDate": {
+        return this.publishedDate;
+      }
+      case "publisher": {
+        return this.publisher;
+      }
+      case "maturityRating": {
+        return this.maturityRating;
+      }
+      case "thumbnail": {
+        return this.thumbnail;
+      }
+      case "isbn": {
+        return this.isbn;
+      }
+      default: {
+        throw new NoSuchFieldException();
+      }
+    }
+  }
+
+  public void setStringField(String fieldName, String value) throws NoSuchFieldException {
+    switch (fieldName) {
+      case "title":
+        this.title = value;
+        break;
+      case "language":
+        this.language = value;
+        break;
+      case "genre":
+        this.genre = value;
+        break;
+      case "description":
+        this.description = value;
+        break;
+      case "infoLink":
+        this.infoLink = value;
+        break;
+      case "pageCount":
+        this.pageCount = value;
+        break;
+      case "publishedDate":
+        this.publishedDate = value;
+        break;
+      case "publisher":
+        this.publisher = value;
+        break;
+      case "maturityRating":
+        this.maturityRating = value;
+        break;
+      case "thumbnail":
+        this.thumbnail = value;
+        break;
+      case "isbn":
+        this.isbn = value;
+        break;
+      default: {
+        throw new NoSuchFieldException();
+      }
+    }
+  }
+
+  public ArrayList<String> getArrayField(String fieldName) throws NoSuchFieldException {
+    if (fieldName.equals("categories")) {
+      return this.categories;
+    } else {
```

Comment:
maybe use an else-if and a noSuchFieldException() again since the user may put in bad input?